jsonrpc/provider/src/provider.ts

```diff
@@ -102,6 +102,21 @@ export class JsonRpcProvider extends IJsonRpcProvider {
     }
   }
 
+  protected onClose(event?: CloseEvent): void {
+    // Anything other than 1000 is an abnormal closure -> also emit an error in this case.
+    if (event && event.code !== 1000) {
```

Comment:
@ganchoradkov keen to get your thoughts on this.

If you think this is too broad of an approach we can instead only check `event.code === 3000` which is what the relay now sends together with `event.reason` when using `useOnCloseEvent=true` and there's an unexpected error/close due to missing projectId, bad JWT etc etc

Reply:
Seems fine as is. Let's capture everything and narrow it down if needed instead of possibly missing important

Reply:
That was my original thinking too, but it seems like the close code is e.g. `1005` if you call `provider.disconnect()` explicitly (also tested in the sample wallet via `core.relayer.provider.disconnect()`), so given that it seems too broad/prone to false positives.

Reply:
I've now changed to `event.code === 3000`.

I think safer to start with the specifically agreed one and work our way outwards if we're not capturing certain aspects as expected.